src/simulationMachine.tsx

```diff
@@ -106,13 +110,26 @@ export const simulationMachine = simModel.createMachine(
         invoke: {
           id: 'proxy',
           src: () => (sendBack, onReceive) => {
-            const receiver = createWindowReceiver({
-              // for some random reason the `window.top` is being rewritten to `window.self`
-              // looks like maybe some webpack replacement plugin (or similar) plays tricks on us
-              // this breaks the auto-detection of the correct `targetWindow` in the `createWindowReceiver`
-              // so we pass it explicitly here
-              targetWindow: window.opener || window.parent,
-            });
+            const serverUrl = new URLSearchParams(window.location.search).get(
+              'server',
+            );
+
+            let receiver: InspectReceiver;
+            if (serverUrl) {
+              const [protocol, ...server] = serverUrl.split('://');
+              receiver = createWebSocketReceiver({
+                protocol: protocol as 'ws' | 'wss',
+                server: server.join('://'),
```

Comment:
This looks somewhat off and could lead to some invalid URL being passed here (or rather to an invalid URL being constructed [here](https://github.com/statelyai/xstate/blob/fd69727fe787196b69f83931ff6944f6e9bdd0ee/packages/xstate-inspect/src/browser.ts#L301)). We might slightly refactor this later but no need to block this PR on this nit-picky comment